### PR TITLE
[LoongArch64] Fix the add overflow check for A=A+A.

### PR DESCRIPTION
**Pull request by sunlijun-610**

### CODE REVIEW

**Comment by sunlijun-610** [comment]
@shushanhf  Could you please review this change？
Thanks！

File: src/coreclr/jit/emitloongarch64.cpp [hunk 1]
```diff
@@ -4918,7 +4918,7 @@ regNumber emitter::emitInsTernary(instruction ins, emitAttr attr, GenTree* dst,
 
         if (needCheckOv)
         {
-            emitIns_R_R_R(INS_or, attr, REG_R21, nonIntReg->GetRegNum(), REG_R0);
+            emitIns_R_R_I(INS_ori, attr, REG_R21, nonIntReg->GetRegNum(), 0);
```
Comment (huoyaoyuan) [inline src/coreclr/jit/emitloongarch64.cpp:4921]: Is this change necessary? Does it have any difference between `OR zr` and `ORI imm#0`?
Which one is used consistently in codegen for LA?

**Comment by sunlijun-610** [comment]
@shushanhf  Could you please review this change？
Thanks！

File: src/coreclr/jit/emitloongarch64.cpp (hunk 1 above)
Comment (sunlijun-610) [reply src/coreclr/jit/emitloongarch64.cpp:4921]: Here `OR zr` and `ORI imm#0` do the same thing, but `ORI imm#0` is used consistently for `$rd=$rn`.

**Review by shushanhf — APPROVED** [review]
LGTM

@jakobbotsch  Could you please review this PR?
Thanks

**Comment by sunlijun-610** [comment]
@jakobbotsch Could you please review this PR?
Thanks a lot!

**Review by jakobbotsch — APPROVED** [review]
